proposals/variant_set_metadata/README.md

```diff
@@ -0,0 +1,246 @@
+there isn't a uniform way to do so. Yes, the author could prefix the name of the `UsdVariantSet` 
+with something like "test" or "__", but every editor would need to respect this prefix.
+A better way to support such a feature on `UsdVariantSet` would be to expose a hidden metadata field.
+This would be consistent with the `UsdObject` API and lends itself to other metadata fields
+that would be applicable for a `UsdVariantSet` like display name.
+
+## Metadata Fields
+
+The following are metadata fields that will be commonly used on a `UsdVariantSet` 
+and will have an explicit API. The fields proposed here will provide the normal accessors 
+and mutators that will be available on a `UsdVariantSet`:
+
+- `comment` - User notes about a variant set
+- `documentation` - Information that can describe the role of the `UsdVariantSet` and its intended uses
+- `displayGroup` - Name to assist with grouping similar `UsdVariantSet`s in editors.
+- `displayName` - Name for the `UsdVariantSet` that can be used in editors
+- `hidden` - Boolean field that can inform an editor if the `UsdVariantSet` should be hidden. 
+    This field will have a fallback value of `false`
+- `variantDisplayNames` - Mapping of variant name to a variant's display name that can be used in editors. 
+    See [Variant Display Names](#variant-display-names) for more implementation details.
+- `variantOrder` - User defined ordering of variants, similar to primOrder and propertyOrder. 
+    See [Variant Ordering](#variant-ordering) for more implementation details.
+- `customData` - User defined dictionary of additional metadata
+
+## Custom Metadata Fields
+
+Similar to other types that allow authoring of metadata fields, `UsdVariantSet` will also 
+allow custom metadata fields that are defined in plugins. These custom fields can be used to 
+author metadata on a `UsdVariantSet` that might be site-specific but should always be applied 
+to the `UsdVariantSet` type.
+
+## Metadata Value Resolution
+
+As described below, a [UsdVariantSet should not be a UsdObject](#usdvariantset-is-not-a-usdobject). 
+But the metadata value resolution process of a `UsdVariantSet` should behave 
+similar to that of `UsdObject`. For most types the resolution process will be 
+"strongest opinion wins" but other, more complicated types like VtDictionary will be 
+correctly resolved. This will slightly complicate the implementation since 
+`UsdVariantSet` will need to provide this value resolution step, 
+but should provide expected results for authored metadata.
+
+## Example
+
+An example showing metadata on variant sets can be found [here](example.usda)
+
+```
+#usda 1.0
+(
+    defaultPrim = "MilkCartonA"
+)
+
+def Xform "MilkCartonA" (
+    kind = "prop"
+    variants = {
+        string modelingVariant = "Carton_Opened"
+        string shadingComplexity = "full"
+        string shadingTest = "milkBrandA"
+    }
+    add variantSets = ["modelingVariant", "shadingComplexity", "shadingTest"]
```

Comment:
Minor note-- can we use `append` instead of `add` here?

Reply:
Thanks for noting that, @nvmkuruc ... even more minor note: if I had to reach for a "default" replacement for `add` I would probably go with `prepend`, because it will order your additions to the front of the list.  In a base asset it likely won't make much difference, but adding variantSets in "stronger referencing layers", you _probably_ intend for them to be stronger than the variantSets coming from across the reference/subLayers.

Reply:
Agreed that's better.

Reply:
I agree, `prepend` is probably the expected behavior. Updated